pom.xml

```diff
@@ -112,8 +112,8 @@
       <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
       <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
-      <maven.compiler.source>1.7</maven.compiler.source>
-      <maven.compiler.target>1.7</maven.compiler.target>
+      <maven.compiler.source>1.8</maven.compiler.source>
+      <maven.compiler.target>1.8</maven.compiler.target>
```

Comment:
From what I've saw from a dev's perspective: from the newest code, If I try to import the source into Eclipse's Neon version, I've got a very single point which requires 1.8 source (at artemis-jdbc-store project):

``
fileLocks.putIfAbsent(fileName, new Object());
``

Setting Java 1.7 version will fail at that point if you set in that way. That's why I've stated to upgrade the version, considering all tests were passed flawlessly and from a server perspective this is a big win. The main point I can see here it's **all** about the client-side (as @clebertsuconic said), especially because the client-side counterparts are dealing with several different libraries versions, different businesses realities and conflicting ways to keep things without any further jar-hell issues.

The thing is: we can see Artemis project being influenced by java 8 and it is a matter of time to have lambda / stream API all over the code in order to improve performance, parallelism and scalability. It's a natural way to get things progressively done from this standpoint.

Considering @clebertsuconic  statement that 1.7 x 1.8 is a general complaint, it's a matter to say how long this compatibility thing will prevail.

The main purpose of this commit it's all about to get things working properly under an IDE environment, it's very good thing when you are directing people to contribute with source code, it's a matter to say how good this is ok to do now.